### ISSUE TITLE
axios.post sends a GET request

### ISSUE THREAD

**Issue by ligne13** [axios/axios#1629]
#### Summary

This code should send a POST request, but sends a GET request ! It happens only when my app is deployed on my production server. But the bug happens before reaching the back-end so I don't understand what is happening and why it is related to the environment. No problem on my local machine.

```
axios.post('/api/admin/tags/', formData)
         .then((response) => {
             console.log(response.data);
         })
         .catch((error) => {
             console.log(error.response.data);
         });
```

#### Context

- axios version: v0.17.1

**Comment by ligne13** [comment]
OK just found the solution ! https://stackoverflow.com/questions/47208440/axios-xmlhttprequest-is-sending-get-instead-of-post-in-production-environment

this was caused the trailing slash at the end of the request URI, which was causing a 301 redirect. The POST request was not visible in the XHR tab of Chrome's debug console.

**Comment by shirbr510** [comment]
this is quite weird cause I just had the opposite case.

my endpoint expects `/api/v1/organizations/` but if I send it `/api/v1/organizations` and it requests a 301 request, it automatically does a get request.

**Comment by fuqi1001** [comment]
Same here, some api routes need tail '/', I just skipped it and it automatically does a GET request

**Comment by Goahnary** [comment]
I am having the same issue today. I have a trailing slash `axios.post('/shop/cart/', data)` and it sends a get request instead of a post.

Package version: `0.18.0`

**Comment by Goahnary** [comment]
So... is this an issue with the middleware that redirects? Or is this an actual issue with axios?

**Comment by KayO-GH** [comment]
With or without a trailing slash, it still sends a GET request. Very frustrating since it causes the request to abort. I am trying to upload a file from a React app to a Django server.

Package version: 0.19.0

**Comment by dzineer** [comment]
Interesting how there is no answer. Did something break in Axios?

**Comment by bishoywagih** [comment]
same here i send the request without / and wit POST request and it changed to Get request

```axios.post('/submit-answer/' + this.questionData.id, {
                answer: this.answer
            })
            .then( response => {
                this.questionData = response.data.detail;
            });
```

**Comment by gajet5** [comment]
After analyzing the problem, I identified three items that can create strange behavior.

1. "/" at the end of the query.
2. The redirection on the server side.
3. automatic URL generation

In my case, there was a third situation. I use the i18n plugin. 

My request:
```
axios({
  method: 'POST',
  url: '/account/login',
  data: {
    email: 'login',
    password: 'password',
  } 
});
```

When prompted, the i18n plugin translated the request with the language in url /en/account/user. After I processed the request on the client side by adding the language to url /en/account/login, the problem went away.

I hope this is helpful.
